File: life.py
import random
import os
import time

from colorama import Fore
from colorama import Style

class cell:

    def __init__(self , i , j , size , body='.' , Alive=False):
        
        self.row = i 
        self.col = j 

        self.size = size

        self.body = body

        # self.Neibours = None

        self.Alive = Alive
        
        self.id = self.col + self.row * 10

    def update(self , Neibour):

        neibour_count = 0

        if self.row > 0:
            if Neibour[self.row - 1][self.col].body == '*':
                neibour_count += 1

        if self.row < self.size -1:
            if Neibour[self.row + 1][self.col].body == '*':
                neibour_count += 1

        if self.col > 0:
            if Neibour[self.row][self.col- 1].body == '*':
                neibour_count += 1


        if self.col < self.size -1 :
            if Neibour[self.row][self.col + 1].body == '*':
                neibour_count += 1




        if self.row > 0 and self.col > 0:
            if Neibour[self.row - 1][self.col -1].body == '*':
                neibour_count += 1

        if self.row < self.size - 1 and self.col < self.size - 1:
            if Neibour[self.row + 1][self.col + 1].body == '*':
                neibour_count += 1

        if self.row < self.size - 1 and self.col > 0:
            if Neibour[self.row + 1][self.col - 1].body == '*':
                neibour_count += 1

        if self.row > 0 and self.col < self.size - 1:
            if Neibour[self.row - 1][self.col + 1].body == '*':
                neibour_count += 1


        if neibour_count < 2:

            self.Alive = False

        elif neibour_count > 3:

            self.Alive = False
        
        elif self.body == '*' and neibour_count == 2:

            self.Alive = True

        elif neibour_count == 3:

            self.Alive = True

        else:
            self.Alive = False
        
        
size = 100

def display(life):
    
    for i in range(size):
        for j in range(size):

            if life[i][j].Alive:
                print(f'{Style.BRIGHT}{Fore.WHITE}{life[i][j].body}', end=' ')
            else:
                print(f'{Style.BRIGHT}{life[i][j].body}', end=' ')

        print()

def update(life):

    for i in range(size):
        for j in range(size):
            
            life[i][j].update(life)


    for i in range(size):
        for j in range(size):
            
            if life[i][j].Alive:

                life[i][j].body = '*'

            else:
                life[i][j].body = '.'


def main():

    life = [['*' for i in range(size)] for j in range(size)]
    

    for i in range(size):
        for j in range(size):

            rand_pop = random.randint(1 , 2)

            if rand_pop == 1:
            
                life[i][j] = cell(i , j , size , body='*' , Alive=True)

            else:
                life[i][j] = cell(i , j , size)


    while True:

        display(life)
        update(life)

        time.sleep(0.1)
        os.system('clear')
        print()
        print()

main()

File: life.js
class cell{

    constructor( i , j , s , w , h , alive ){
        
        this.row = i;
        this.col = j;

        // this.row = i;
        // this.col = j;

        this.s = s;

        // this.size = w;
        // this.size = h;
        this.w = w;
        this.h = h;

        this.Alive = alive;
        this.stay = true;

        this.body = "*";

    }

    make(){
        stroke(255);
        strokeWeight(0.1);
        
        if( this.stay ){
            
            // fill(0 , 255 , 0);
            fill(0 , 255 , 0);
        }

        else{
            // fill(255 , 0 , 0);
            fill(0 , 0 , 0);

        }
        rect(this.row*this.s , this.col*this.s , this.s-1 , this.s-1);
    }

    add_neibour(Neibour){
        
        this.neibour_count = 0

        if( this.row > 0){
            if( Neibour[this.row - 1][this.col].Alive ){
            
                this.neibour_count += 1
            }
        }

        if( this.row < this.w -1){
                if( Neibour[this.row + 1][this.col].Alive ){

            
                this.neibour_count += 1
            }
        
        }

        if( this.col > 0){

            if( Neibour[this.row][this.col- 1].Alive ){
            
                this.neibour_count += 1
            }
        }


        if( this.col < this.h -1 ){

                if( Neibour[this.row][this.col + 1].Alive ){
            
                this.neibour_count += 1
            }
        }

        if( this.row > 0 && this.col > 0){
                if( Neibour[this.row - 1][this.col -1].Alive ){
            
                this.neibour_count += 1
            }
        
        }

        if( this.row < this.w - 1 && this.col < this.h - 1){
                if( Neibour[this.row + 1][this.col + 1].Alive ){
            
                this.neibour_count += 1
            }
        }

        if( this.row < this.w - 1 && this.col > 0){
                if( Neibour[this.row + 1][this.col - 1].Alive ){
            
                this.neibour_count += 1
            }
        
        }

        if( this.row > 0 && this.col < this.h - 1){
            if( Neibour[this.row - 1][this.col + 1].Alive ){
            
                this.neibour_count += 1
            }
        
        }


        if( this.neibour_count < 2){

            this.Alive = false
        }


        else if( this.neibour_count > 3){

            this.Alive = false
        }
        

        else if( this.Alive  && this.neibour_count == 2){

            this.Alive = true
        }


        else if( this.neibour_count == 3){

            this.Alive = true
        }

        else{

            this.Alive = false
        }

     
    }

}